packages/relay/src/lib/eth.ts

```diff
@@ -1897,17 +1897,24 @@ export class EthImpl implements Eth {
     return logs;
   }
 
-  private async getAccountLatestEthereumNonce(address: string, requestIdPrefix?: string) {
-    // check if address is a valid contract then get ethereumNonce from consensus node until HIP 729 is implemented
-    const validContract = await this.mirrorNodeClient.isValidContract(address, requestIdPrefix);
-    if (validContract)  {
+  private async getAccountLatestEthereumNonce(address: string, requestId?: string) {
+    const cachedResponse: any = this.mirrorNodeClient.getIsValidContractCache(address);
+
+    // address is a valid contract
+    if (cachedResponse) {
       return EthImpl.oneHex;
-    } 
-    
-    // get latest ethereumNonce from mirror node account API
-    const mirrorAccount = await this.mirrorNodeClient.getAccount(address, requestIdPrefix);
-    if (mirrorAccount?.ethereum_nonce) {
-      return EthImpl.numberTo0x(mirrorAccount.ethereum_nonce);
+    }
+
+    const accountData = await this.mirrorNodeClient.getAccount(address, requestId);
+    if (accountData) {
+      if (!accountData.ethereum_nonce) {  // if nonce > 0 then the entity cannot be a contract
```

Comment:
Q: just to confirm this works as expected when nonce is 0. Can add a test to be sure.

Reply:
Confirmed offline